src/shared/utilities/vsCodeUtils.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+export const localize = nls.loadMessageBundle()
+
+import { ErrorOrString, getLogger, Logger } from '../logger'
+
+// ------- Experimental combined output channel & traditional logger  --------------
```

Comment:
I like the idea, but I don't think this section is mature/complete enough yet to merge into develop. If you find it's the only thing holding back the review, consider breaking it out and PRing it after this one.

Reply:
Tests added.